cognite/pygen/_query/interface.py

```diff
@@ -277,28 +331,37 @@ def _execute_aggregation(
         if metric_aggregates and histogram_aggregates:
             raise ValueError("Cannot mix metric and histogram aggregations")
 
+        view = self._get_view(view_id)
+        instance_types = self._get_instance_types(view)
         if metric_aggregates and group_by is not None:
-            group_by_result = self._client.data_modeling.instances.aggregate(  # type: ignore[call-overload]
-                view=view_id,
-                group_by=group_by,
-                aggregates=metric_aggregates,
-                query=query,
-                properties=search_properties,
-                filter=filter,
-                limit=limit or AGGREGATION_LIMIT,
-            )
-            return self._grouped_metric_aggregation_to_dict(group_by_result)
+            all_group_results: dict[str, dict[str, Any]] = {}
+            for instance_type in instance_types:
+                group_by_result = self._client.data_modeling.instances.aggregate(  # type: ignore[call-overload]
+                    instance_type=instance_type,
+                    view=view_id,
+                    group_by=group_by,
+                    aggregates=metric_aggregates,
+                    query=query,
+                    properties=search_properties,
+                    filter=filter,
+                    limit=limit or AGGREGATION_LIMIT,
+                )
+                self._update_group_aggregation(group_by_result, all_group_results)
+            return list(all_group_results.values())
         elif metric_aggregates:
-            metric_results = self._client.data_modeling.instances.aggregate(
-                view_id,
-                aggregates=metric_aggregates,
-                query=query,
-                properties=search_properties,
-                filter=filter,
-                limit=limit or AGGREGATION_LIMIT,
-            )
-            return self._metric_aggregation_to_dict(metric_results)
-
+            metric_dict: dict[str, dict[str, float | int | None]] = {}
```

Comment:
This can be a candidate for you to have a TypeAlias with this definition, might be useful in the future ;)

Reply:
Added a more descriptive name